### ISSUE TITLE
Incorrect results on carvana?

### ISSUE THREAD

**Issue by MaxEAB** [milesial/Pytorch-UNet#417]
Hi, Thanks for this code. 

As a sanity check, I tried using the pre-trained carvana model to test the segmentation on the image below. After thresholding the predicted mask at 0.5, I see the following. 

![00d3de919651_05](https://user-images.githubusercontent.com/103768359/210847580-8f068ae1-93b4-4e71-9095-c7a23c47adb4.jpg)
![test_mask_cls1](https://user-images.githubusercontent.com/103768359/210847585-c3a495ec-0f9b-461b-969f-9cf805b9db9c.png)

I loaded the model as, 
`torch.hub.load('milesial/Pytorch-UNet', 'unet_carvana', pretrained=True, scale=1.0)`

Why isn't the segmentation clean? How do I fix this?

**Comment by milesial** [comment]
Can you show the exact filename for that image input? I'll try on my side

**Comment by MaxEAB** [comment]
00d3de919651_05.jpg from test.zip [here](https://www.kaggle.com/c/carvana-image-masking-challenge/data?select=test.zip).

**Comment by milesial** [comment]
I get this with the 0.5 scale
![image](https://user-images.githubusercontent.com/30204471/211139398-7b39bd2f-03fc-443c-a299-356bb5ffbbc7.png)

and this with the 1.0 scale
![image](https://user-images.githubusercontent.com/30204471/211139414-1407adcd-c05c-4a23-84e3-71ba05e8a3fc.png)

So not perfect but I can't reproduce yours.

What do you mean by "After thresholding the predicted mask at 0.5"? For the pretrained network there is 2 classes so you should take the argmax, there is not thresholding to do.

**Comment by MaxEAB** [comment]
Hi, Thanks for checking. I switched to argmax and this is what I get:

![test_arg-max_mask_cls](https://user-images.githubusercontent.com/103768359/211417827-2d99762c-e92c-4014-a30a-a3de39d78e89.png)

Would you be able to share the code that you used to test this?

**Comment by milesial** [comment]
I just used the predict.py script. Are you not using that?

`python predict.py -m model.pth -i image.jpg --scale 0.5`

**Comment by MaxEAB** [comment]
Hi, Thanks. I was using my own code to evaluate, and the image scaling was incorrect. Now I get the same results as you.

Which are the two classes trained? -- car and background?

**Comment by milesial** [comment]
Good to hear.
Yes, 0 is background, 1 is car.
